normalizer_lib.py

```diff
@@ -1,205 +1,224 @@
-        if difference(acceptor(token), valid_token).num_states() == 0:
-            returned.append(token)
-        else:
-            returned.append("<UNK>")
-    return " ".join(returned)
 
+    #Discard invalid tokens
+    # e.g. "how are you today товарищ?" -> "how are you today?"
 
-def pass_only_valid_sentences(string: str) -> str:
-    """Replaces invalid sentences.
 
-    Args:
-        string: A line from a corpus.
+    def pass_only_valid_tokens(self, string) -> List[str]:
+        """Replaces invalid strings in a sentence.
 
-    Returns:
-        The line or the line replaced with <SENTENCE_REJECTED>.
-    """
-    valid_token = INITIAL_PUNCTUATION.ques + Verbalizable.verbalizable() + FINAL_PUNCTUATION.ques
-    remove_extra_whitespace = (string @ REMOVE_EXTRA_WHITESPACE).string()
-    split_string = remove_extra_whitespace.split(" ")
-    for token in split_string:
-        if difference(acceptor(token), valid_token).num_states() != 0:
-            return "<SENTENCE_REJECTED>"
-    return string
-
-
-# Detach punctuation from words
-# e.g. "Who are you?" -> "Who are you ?"
-
-NON_GRAPHEME_PUNCT = difference(PUNCTUATION, GRAPHEMES)
-
-SEPARATE_PUNCTUATION = (
-    cdrewrite(
-        pynutil.insert(" "),
-        union("[BOS]", byte.SPACE) + union(NON_GRAPHEME_PUNCT),
-        (union(PUNCTUATION, Verbalizable.verbalizable()) +
-         union("[EOS]", byte.SPACE, Verbalizable.verbalizable())),
-        SIGMA_STAR) @
-    cdrewrite(
-        pynutil.insert(" "),
-        union(Verbalizable.verbalizable(), PUNCTUATION),
-        PUNCTUATION + union("[EOS]", byte.SPACE, PUNCTUATION),
-        SIGMA_STAR))
+        Args:
+            string: A line from a corpus.
 
+        Returns:
+            The line, where invalid tokens have been replaced with <UNK>.
+        """
+        valid_token = (self.initial_punctuation.ques +
+                       self.verbalizable() +
+                       self.final_punctuation.ques)
+        returned: List[str] = []
+        remove_extra_whitespace = (string @ self.remove_extra_whitespace()).optimize().string()
+        split_string = remove_extra_whitespace.split(" ")
+        for token in split_string:
+            if difference(acceptor(token), valid_token).num_states() == 0:
+                returned.append(token)
+            else:
+                returned.append("<UNK>")
+        return " ".join(returned)
+
+
+    def pass_only_valid_sentences(self, string) -> List[str]:
+        """Replaces invalid sentences.
+
+        Args:
+            string: A line from a corpus.
+
+        Returns:
+            The line or the line replaced with <SENTENCE_REJECTED>.
+        """
+        valid_token = (self.initial_punctuation.ques +
+                       self.verbalizable() +
+                       self.final_punctuation.ques)
+        remove_extra_whitespace = (string @ self.remove_extra_whitespace()).optimize().string()
+        split_string = remove_extra_whitespace.split(" ")
```

Comment:
Some repetition from the previous method, consider adding a helper function which contains the shared logic.

Reply:
I ended up not needing `valid_token`, since that is pretty easily defined by `self.verbalizable()`, but I also put the splitting into tokens in a helper function to tidy things up.